lgc/patch/ShaderInputs.cpp

```diff
@@ -331,6 +331,12 @@ void ShaderInputs::fixupUses(Module &module, PipelineState *pipelineState) {
 
     ShaderStage stage = getShaderStage(&func);
     ShaderInputsUsage *inputsUsage = getShaderInputsUsage(stage);
+
+    bool useWorkgroupIds[3] = {true};
+    bool useWholeWorkgroupId = true;
+    if (stage == ShaderStageCompute)
+      memset(useWorkgroupIds, false, sizeof(bool));
```

Comment:
I recommend a slightly different and more robust algorithm:

* Start out with all `useXxx` values set to `false`
* When encountering an `extractelement` user with constant index, set the corresponding entry in the array to `true`
* When encountering any other user, set the `useWholeWorkgroupId` to `true`. This catches the general case, e.g. where the workgroup ID is directly stored to memory, or is indexed dynamically.
* Adjust the attribute adding logic: wrap it in a check for ShaderStageCompute, and change each individual check to `!useWholeWorkgroupId && !useWorkgroupIds[0]` etc.

Reply:
Thank you for the elaborate comments and making the algorithm robust. Fixed.